sktime/annotation/hmm.py

```diff
@@ -0,0 +1,275 @@
+# -*- coding: utf-8 -*-
+"""
+HMM Annotation Estimator.
+
+Implements a basic Hidden Markov Model (HMM) as an annotation estimator.
+To read more about the algorithm, check out the wikipedia page:
+(https://en.wikipedia.org/wiki/Hidden_Markov_model
+"""
+from math import isclose
+from typing import Tuple
+
+import numpy as np
+
+from sktime.annotation.base._base import BaseSeriesAnnotator
+
+__author__ = ["miraep8"]
+__all__ = ["HMM"]
+
+
+class HMM(BaseSeriesAnnotator):
+    """Implements a simple HMM fitted with viterbi algorithm.
+
+    The HMM annotation estimator facilitates the appilcation of
+    the vertibi algorithm to fit a sequence of 'hidden state' class
+    annotations (represented by an array of integers the same size
+    as the observation) to a sequence of observations.
+
+    This is done by finding the most likely path given the emission
+    probabilities - (ie the probability that a particular observation
+    would be generated by a given hidden state), the transition prob
+    (ie the probability of transitioning from one state to another or
+    staying in the same state) and the initial probabilities - ie the
+    belief of the probability distribution of hidden states at the
+    start of the observation sequence).
+
+    Parameters
+    ----------
+    emission_funcs : a list of functions, which should be properly
```

Comment:
must be more precise. What signature do the functions have?
Should also probably be callable.